### ISSUE TITLE
Some questions for image mean subtraction

### ISSUE THREAD

**Issue by cloudseasail** [tensorpack/tensorpack#1072]
1. for per_pixel_mean(), it is better to avoid using test_files.  It is unfair to use test_files for training datas which apply subtle distribution from test_files,  and I am not sure if it will affect final test eval
2. for test, it is better to use image mean retrieved from training images.  and it is better to save the image mean in the model for further usage at prediction.  wondering is there easy way to save/restore it along with model?

```
def get_per_pixel_mean(self):
        """
        Returns:
            a mean image of all (train and test) images of size 32x32x3
        """
        train_files, test_files, _ = get_filenames(self.dir, self.cifar_classnum)
        all_imgs = [x[0] for x in read_cifar(train_files + test_files, self.cifar_classnum)]
        arr = np.array(all_imgs, dtype='float32')
        mean = np.mean(arr, axis=0)
        return mean
```

**Comment by ppwwyyxx** [comment]
1. That makes sense. I don't believe it will have visible effect to accuracy since the difference is < 0.5 out of image range of 255. But using the training mean is the legitimate thing to do.

2. Tensorpack saves with tensorflow's saver, which saves variables in the graph only. If you want to save anything else to the checkpoint, you can create a variable in the graph with this value. This is a tensorflow issue and unrelated to tensorpack.

**Comment by ppwwyyxx** [comment]
1 was fixed now.

For 2, an alternative way to perhaps simplify prediction is to do the mean subtraction in the graph, and just call the same `build_graph` function in inference.
